base/src/main/java/com/austindroids/austinfeedsme/data/EventsRepository.kt

```diff
@@ -18,55 +18,25 @@ package com.austindroids.austinfeedsme.data
 
 import io.reactivex.Observable
 import io.reactivex.Single
-import io.reactivex.SingleEmitter
-import io.reactivex.SingleOnSubscribe
-import io.reactivex.subjects.PublishSubject
-import javax.inject.Singleton
 
-@Singleton
-class EventsRepository(private val eventsRemoteDataSource: EventsDataSource) : EventsDataSource, RxEventsDataSource {
+open class EventsRepository(private val eventsRemoteDataSource: EventsDataSource) : RxEventsDataSource {
 
-    val eventsSubject : PublishSubject<List<Event>> = PublishSubject.create()
-
-    override fun getEvents(callback: EventsDataSource.LoadEventsCallback, onlyFood: Boolean) {
-        eventsRemoteDataSource.getEvents(object : EventsDataSource.LoadEventsCallback {
-            override fun onEventsLoaded(events: List<Event>) {
-                callback.onEventsLoaded(events)
-            }
-
-            override fun onError(error: String) {
-                callback.onError(error)
-            }
-        }, onlyFood)
-    }
-
-    override fun saveEvent(eventToSave: Event, callback: EventsDataSource.SaveEventCallback?) {
-        eventsRemoteDataSource.saveEvent(eventToSave, object : EventsDataSource.SaveEventCallback {
-            override fun onEventSaved(success: Boolean) {
-                callback?.onEventSaved(success)
-            }
-
-            override fun onError(error: String) {
-                callback?.onError(error)
-            }
-        })
-    }
-
-    override fun getEventsRX(onlyFood: Boolean): Observable<List<Event>> {
-        eventsRemoteDataSource.getEvents(object : EventsDataSource.LoadEventsCallback {
-            override fun onEventsLoaded(events: List<Event>) {
-                eventsSubject.onNext(events)
-            }
+    override fun getEventsRX(): Observable<List<Event>>? {
+        return Observable.create {
```

Comment:
So this is where `it` is coming from. If this was my project, I wouldn't let this go in unless it was explicitly named to `emitter`.